driver/gl/window.go

```diff
@@ -60,6 +60,34 @@ func (w *window) SetFullScreen(full bool) {
 	})
 }
 
+func (w *window) CenterOnScreen() {
+	runOnMainAsync(func() {
+		// get window dimensions in pixels
+		monitor := getMonitorForWindow(w.viewport)
+		monMode := monitor.GetVideoMode()
+
+		// these come into play when dealing with multiple monitors
+		monX, monY := monitor.GetPos()
+		
+		// get current size of content inside the window
+		winContentSize := w.Content().MinSize()
+		// get current window dimensions in pixels
+		viewWidth, viewHeight := w.viewport.GetSize()
+
+		// take the larger of the window size and the content
```

Comment:
I wonder - should we be centering if it’s hidden? It would avoid complexity but may cause a race condition...

Reply:
What race condition would it cause?  My thinking was if you created a window and wanted it displayed in the center of the screen, this is the way to do it instead of displaying it in the upper corner and having the user see it become centered.

Reply:
I think you are right.
The potential race was if it recorded the fact it wanted to be centered but then applied it after the window became visible. Probably best the way you approached it :)